File: solutions/fsharp/raindrops/1/Raindrops.fs
module Raindrops

let raindrops = [(3, "Pling"); (5, "Plang"); (7, "Plong")] 

let fetch number str (modulo, text) : string = 
    if (number % modulo = 0) then str + text else str

let convert (number: int): string = 
    let mutable str = ""
    for raindrop in raindrops do
        str <- fetch number str raindrop

    if (str <> "") then str else number.ToString()

File: solutions/csharp/raindrops/1/Raindrops.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Raindrops
{
    public static string Convert(int number)
    {
        var results =
            new List<(int, string)> {(3, "Pling"), (5, "Plang"), (7, "Plong")}
                .Where(f => number % f.Item1 == 0)
                .Select(f => f.Item2)
                .ToList();

        return results.Any() ? string.Concat(results) : number.ToString();
    }
}